mirdata/annotations.py

```diff
@@ -16,91 +84,149 @@ class BeatData(Annotation):
     """BeatData class
 
     Attributes:
-        times (np.ndarray): array of time stamps (as floats) in seconds
-            with positive, strictly increasing values
-        positions (np.ndarray or None): array of beat positions (as ints)
-            e.g. 1, 2, 3, 4
+        times (np.ndarray): array of time stamps with positive,
+            strictly increasing values
+        time_unit (str): time unit, one of TIME_UNITS
+        positions (np.ndarray): array of beat positions in the format
+            of position_unit. For all units, values of 0 indicate beats which
+            fall outside of a measure.
+        position_unit (str): beat position unit, one of BEAT_POSITION_UNITS
+        confidence (np.ndarray): array of confidence values
+        confidence_unit (str): confidence unit, one of CONFIDENCE_UNITS
 
     """
 
-    def __init__(self, times, positions=None):
+    def __init__(
+        self,
+        times,
+        time_unit,
```

Comment:
I wanted to have it not default on purpose so that when the loader is defined, we have to think about what the units should be. (to avoid very evil secret bugs when we use them!)